internal/lib/lib.go

```diff
@@ -1885,3 +1888,25 @@ func GetIstioPKIProfileName() string {
 func GetIstioWorkloadCertificateName() string {
 	return "istio-workload-" + GetClusterName() + "-" + utils.GetAKONamespace()
 }
+
+var istioCertSet sets.String
+
+func updateIstioCertSet(s string) {
+	if istioCertSet == nil {
+		istioCertSet = sets.NewString()
+	}
+	istioCertSet.Insert(s)
+}
+
+func GetIstioCertSet() sets.String {
+	return istioCertSet
+}
+
+func IsChanClosed(ch <-chan struct{}) bool {
+	select {
+	case <-ch:
```

Comment:
This what we have to do to check whether the channel is closed, right?
```
_, ok := <-ch
return ok
```

Reply:
i think this will cause the go routine to wait until some data is sent over the channel.